test/unitTests/core/zoe/test-isOfferSafe.js

```diff
@@ -0,0 +1,236 @@
+      { rule: 'haveExactly', amount: assays[0].make(2) },
+      { rule: 'wantExactly', amount: assays[1].make(3) },
+      { rule: 'wantExactly', amount: assays[2].make(3) },
+    ];
+    const amounts = [assays[0].make(2), assays[1].make(3), assays[2].make(3)];
+    t.ok(isOfferSafeForPlayer(assays, rules, amounts));
+  } catch (e) {
+    t.assert(false, e);
+  } finally {
+    t.end();
+  }
+});
+
+// The user gets more than they wanted - wantExactly
+test('isOfferSafeForPlayer - more than wantExactly', t => {
+  try {
+    const { assays } = setup();
+    const rules = [
+      { rule: 'wantExactly', amount: assays[0].make(2) },
+      { rule: 'wantExactly', amount: assays[1].make(3) },
+      { rule: 'wantExactly', amount: assays[2].make(4) },
+    ];
+    const amounts = [assays[0].make(5), assays[1].make(6), assays[2].make(8)];
+    t.ok(isOfferSafeForPlayer(assays, rules, amounts));
+  } catch (e) {
+    t.assert(false, e);
+  } finally {
+    t.end();
+  }
+});
+
+// The user gets more than they wanted - wantAtLeast
+test('isOfferSafeForPlayer - more than wantAtLeast', t => {
+  try {
+    const { assays } = setup();
+    const rules = [
+      { rule: 'wantAtLeast', amount: assays[0].make(2) },
+      { rule: 'wantAtLeast', amount: assays[1].make(3) },
+      { rule: 'wantAtLeast', amount: assays[2].make(4) },
+    ];
+    const amounts = [assays[0].make(5), assays[1].make(6), assays[2].make(8)];
+    t.ok(isOfferSafeForPlayer(assays, rules, amounts));
+  } catch (e) {
+    t.assert(false, e);
+  } finally {
+    t.end();
+  }
+});
+
+// The user gets refunded more than what they put in
+test('isOfferSafeForPlayer - more than haveExactly', t => {
+  try {
+    const { assays } = setup();
+    const rules = [
+      { rule: 'haveExactly', amount: assays[0].make(2) },
+      { rule: 'haveExactly', amount: assays[1].make(3) },
+      { rule: 'haveExactly', amount: assays[2].make(4) },
+    ];
+    const amounts = [assays[0].make(5), assays[1].make(6), assays[2].make(8)];
+    t.ok(isOfferSafeForPlayer(assays, rules, amounts));
+  } catch (e) {
+    t.assert(false, e);
+  } finally {
+    t.end();
+  }
+});
+
+// The user gets less than what they wanted - wantExactly
+test('isOfferSafeForPlayer - less than wantExactly', t => {
+  try {
+    const { assays } = setup();
+    const rules = [
+      { rule: 'haveExactly', amount: assays[0].make(2) },
+      { rule: 'wantExactly', amount: assays[1].make(3) },
+      { rule: 'wantExactly', amount: assays[2].make(5) },
+    ];
+    const amounts = [assays[0].make(0), assays[1].make(2), assays[2].make(1)];
+    t.notOk(isOfferSafeForPlayer(assays, rules, amounts));
+  } catch (e) {
+    t.assert(false, e);
+  } finally {
+    t.end();
+  }
+});
+
+// The user gets less than what they wanted - wantAtLeast
+test('isOfferSafeForPlayer - less than wantExactly', t => {
+  try {
+    const { assays } = setup();
+    const rules = [
+      { rule: 'haveExactly', amount: assays[0].make(2) },
```

Comment:
There are several tests where all the amounts are in violation. How about at least one where only one amount is?

Reply:
Yup, added